### PR TITLE
Migrate away from react-bootstrap modals (ApproveExperienceModal)

### PR DESCRIPTION
**Pull request by AleksandrKurilov**
Migrate ApproveExperienceModal.js

Pull-Request for `paretOS`

## Description
- Migrate ApproveExperienceModal.js to MUI Dialog
- Use react-hook-form

## Relates to
- #87 

## Reviewers
- @mikhael28 

## Screenshots / other info
Before

After

### CODE REVIEW

**Comment by AleksandrKurilov** [comment]
I noticed that the modal doesn't close after clicking "Request Changes". It is correct?

**Comment by mikhael28** [comment]
@AleksandrKurilov it certainly should close - it fires an API request which marks the achievement item as being 'under review', and I believe also sends an email to the athlete letting them know there is an item ready for feedback.

File: src/learn/ApproveExperienceModal.js
Comment (mikhael28) [inline src/learn/ApproveExperienceModal.js:106]: @AleksandrKurilov can you please add this button back in? Right now, you only have the `Request Revisions` button enabled, and we also need to be able to 'Approve' or markComplete.

File: src/learn/ApproveExperienceModal.js
Comment (mikhael28) [inline src/learn/ApproveExperienceModal.js:140]: This is not working sadly - the requestRevisions is not correctly reflected, meaning that when I, as a mentor, sign in to look at my mentees work and 'request revisions' - while, my local environment shows that they were sent for revisions, it's not the case.

What you need to do: can you create a new user, as a 'mentor' and then tell me what the email is? Then I will 'match you' as a mentor with them, and you will be able to test the interplay between the two.

**Review by mikhael28 — CHANGES_REQUESTED** [review]

File: src/learn/ApproveExperienceModal.js
Comment (AleksandrKurilov) [reply src/learn/ApproveExperienceModal.js:140]: @mikhael28 ok, then could you please appoint fieldyslap@yandex.ru as a mentor and fieldyslap@inbox.ru as a mentee

Comment (AleksandrKurilov) [reply src/learn/ApproveExperienceModal.js:140]: @mikhael28 Hi, Did you match users fieldyslap@yandex.ru as a mentor and fieldyslap@inbox.ru as a mentee? I checked yesterday both and I didn't see any changed. Maybe I do something wrong.

Comment (jayeclark) [reply src/learn/ApproveExperienceModal.js:140]: Have you created those users already? I looked quickly in the directory to try and match them, but didn’t see them… I will take a look again! If you’d rather just use an existing generic test set of users, email me at jay@jayeclark.dev or hit me up on Discord and I’ll send you the credentials! (Will be at my PC, where I have the credentials info stored, in about an hour.)

Comment (AleksandrKurilov) [reply src/learn/ApproveExperienceModal.js:140]: @jayeclark Yes, these users are already registered. I will email you. Thank you.

**Comment by AleksandrKurilov** [comment]
@mikhael28  Hey, I finally finished this. Sorry, I was a bit bussy on my work.

image

File: src/learn/ApproveExperienceModal.js
Comment (jayeclark) [reply src/learn/ApproveExperienceModal.js]: @mikhael28 I've tested this issue locally on main and on paret0.com, and requestRevisions isn't 'working' on either of those. However, on all three (live site, local main, this PR branch) it's caused by an issue with the logic that governs how experience is displayed & updated, not with the actual updating of the database via the modal dialog. 

Basically: it's a valid bug but unrelated to @AleksandrKurilov's work converting the modals to MUI and the changes that need to be made are in files this PR doesn't touch.

File: src/learn/ApproveExperienceModal.js
Comment (jayeclark) [reply src/learn/ApproveExperienceModal.js:106]: Resolving, I've verified this has been added back in.

**Comment by jayeclark** [comment]
@mikhael28 this looks good to merge to me, just waiting on your final review - see my comment in the review about requestRevisions and the UI. I should be able to resolve the merge conflicts on my end when merging in.

**Review by mikhael28 — APPROVED** [review]

**Comment by mikhael28** [comment]
It looks good, but sadly I think @AleksandrKurilov needs to resolve the merge conflicts himself - I don't think @jayeclark has push access to his repository. Once the merge is resolved, we will merge in. I've been busy at work myself @AleksandrKurilov , thanks for your patience. Where are you working right now?

**Comment by jayeclark** [comment]
🙌 Merge Conflicts Resolved 🙌

**Pull request closed without merging**
